controller.py

```diff
@@ -215,6 +256,51 @@ def search(self, location, live=True):
             )
             return Response(body, 200, headers)
 
+    def libraries(self):
+        # Return a specific set of information about all libraries; this generates the library list in the admin interface,
+        libraries = []
+        all = self._db.query(Library).order_by(Library.name)
+        for library in all:
+            uuid = library.internal_urn.split("uuid:")[1]
+            libraries += [dict(
+                    id=library.id,
+                    name=library.name,
+                    short_name=library.short_name,
+                    uuid=uuid,
+                    registry_stage=library.registry_stage,
+                    library_stage=library.library_stage,
+                )]
+        return dict(libraries=libraries)
+
+    def library_details(self, uuid):
+        # Return complete information about one specific library; this is triggered when an admin opens the detail page for a library.
+        library = self.library_for_request(uuid)
```

Comment:
Wherever you call `library_for_request` you need to check to see if you got a ProblemDetail, and return the ProblemDetail immediately if so.